homeassistant/components/actiontec/device_tracker.py

```diff
@@ -33,43 +29,40 @@
 )
 
 
-def get_scanner(hass, config):
+def get_scanner(
+    hass: HomeAssistant, config: ConfigType
+) -> ActiontecDeviceScanner | None:
     """Validate the configuration and return an Actiontec scanner."""
     scanner = ActiontecDeviceScanner(config[DOMAIN])
     return scanner if scanner.success_init else None
 
 
-Device = namedtuple("Device", ["mac", "ip", "last_update"])
-
-
 class ActiontecDeviceScanner(DeviceScanner):
     """This class queries an actiontec router for connected devices."""
 
-    def __init__(self, config):
+    def __init__(self, config: ConfigType) -> None:
         """Initialize the scanner."""
         self.host = config[CONF_HOST]
         self.username = config[CONF_USERNAME]
         self.password = config[CONF_PASSWORD]
```

Comment:
Assign proper types to these fields. `config` is `dict[str, Any]` so here they're all `Any`.
```suggestion
        self.host: str = config[CONF_HOST]
        self.username: str = config[CONF_USERNAME]
        self.password: str = config[CONF_PASSWORD]
```